### PR TITLE
GA-59

### PR DESCRIPTION
**Pull request by Josh-HMCTS**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/GA-59

### Change description ###

Add comparison function for saving/updating user with GA roles

### CODE REVIEW

File: src/users/containers/manage-user/manage-user.component.ts [hunk 1]
```diff
@@ -139,8 +171,27 @@ export class ManageUserComponent implements OnInit, OnDestroy {
...
+      ...this.updatedUser
+    };
+    if (value.roles.includes('pui-case-manager')) {
+      value.roles = [...value.roles, ...AppConstants.CCD_ROLES];
+    }
+    value = {
+      ...value,
+      resendInvite: this.resendInvite
+    };
+    console.log(`values pre comparison: ${JSON.stringify(value)}`);
```
Comment (shaed-parkar) [inline src/users/containers/manage-user/manage-user.component.ts:185]: remove console log

**Review by shaed-parkar — COMMENTED** [review]
few changes. but looks good

File: api/retrieveAccessTypes/accessTypesComparison.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      processedAccessTypes.push({
+        jurisdictionId,
+        organisationProfileId,
+        accessTypeId,
+        enabled: accessType.accessDefault
+      });
+    }
+  });
+
+  console.log(processedAccessTypes);
```
[the remark this answers is not in the document]
Comment (Josh-HMCTS) [reply api/retrieveAccessTypes/accessTypesComparison.ts]: removed

File: api/retrieveAccessTypes/index.ts [hunk 3]
```diff
@@ -35,10 +36,46 @@ export async function handleRetriveAccessTypes(req: Request, res: Response) {
         'accessTypes': [
           {
             'organisationProfileId': 'SOLICITOR_PROFILE',
-            'accessTypeId': 'default',
+            'accessTypeId': '5151',
```
[the remark this answers is not in the document]
Comment (Josh-HMCTS) [reply api/retrieveAccessTypes/index.ts]: merged, should work with this stub data

File: src/users/containers/manage-user/manage-user.component.ts (hunk 1 above)
Comment (Josh-HMCTS) [reply src/users/containers/manage-user/manage-user.component.ts]: removed logs

**Review by ben-james-hmcts — COMMENTED** [review]
My only concern here is the use of 'accessType' for variables. I have a similar issue on my current branch. I'm wondering if we have a discussion at some point to decide how we should be labelling anything access type related. For example, should we always refer to them as either user or org specifically?

**Pull request closed without merging**